### ISSUE TITLE
Site Editor: editing a page and setting a password: users can't see the password they are entering

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#52607]
### Description

Please see this comment: https://github.com/WordPress/gutenberg/issues/52605#issuecomment-1634393012

Not sure showing asterisks while users are entering the password they want to use to protect a page is a good UI / User experience.

Unless I'm missing something, seems to me this should be fixed for WP 6.3 as it's a less than ideal UI to release to the public.

Cc @annezazu @tellthemachines for some kind help with triaging ❤️ 

### Step-by-step reproduction instructions

- Edit a page in the Site editor.
- Try to set a password to protect the page.
- Observe that all you can see while entering your password is a nice series of asterisks.

### Screenshots, screen recording, code snippet

<img width="689" alt="asterisks" src="https://github.com/WordPress/gutenberg/assets/1682452/e574ce97-d319-4512-86b3-c8b1a609db33">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by tellthemachines** [comment]
Thanks @afercia ! Added to the 6.3 board.

**Comment by tellthemachines** [comment]
I think the problem is we're using an input of type `password` when we should be using `text`.

**Comment by afercia** [comment]
> I think the problem is we're using an input of type password when we should be using text.

Yes, that is what I said in the comment linked at the beginning of this issue description:

> Please see this comment: https://github.com/WordPress/gutenberg/issues/52605#issuecomment-1634393012

Thanks for the quick triaging and fix ❤️

**Comment by annezazu** [comment]
Thank you for reporting and the quick fix, folks!
